client/src/components/Search.js

```diff
@@ -91,9 +90,14 @@ const Search = (props) => {
               onChange={(_, newValue) => {
                 setState(() => ({ ...state, departureCity: newValue }));
               }}
-              onInputChange={async (_, newInputValue) => {
-                const response = await getCityName(newInputValue);
-                setOptionsDeparture(response);
+              onInputChange={(_, newInputValue) => {
+                if (timeoutRef1.current) {
+                  clearTimeout(timeoutRef1.current);
+                }
+                timeoutRef1.current = setTimeout(async () => {
+                  const response = await getCityName(newInputValue);
+                  setOptionsDeparture(response);
+                }, 500);
```

Comment:
Do you think 5 milliseconds is enough time? It's 5 milliseconds after the last time they had input change right?

Reply:
It's 500 😅 . That's a good standard for debouncing, but it should always be played around with to see what feels best.